drivers/periph_common/timer.c

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers
+ * @{
+ *
+ * @file
+ * @brief       Shared peripheral timer code
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include "periph/timer.h"
+
+#ifndef PERIPH_TIMER_PROVIDES_SET
+int timer_set(tim_t dev, int channel, unsigned int timeout)
+{
+    return timer_set_absolute(dev, channel, timer_read(dev) + timeout);
```

Comment:
I might also be good to add a safety check on dev before this line. Something like:
```c
if (dev >= TIMER_NUMOF) {
        return -1;
    }
```
I saw that in one of the other impl that is removed in this PR.